hoodie-hive/src/main/java/com/uber/hoodie/hive/HiveSyncTool.java

```diff
@@ -61,15 +64,38 @@ public HiveSyncTool(HiveSyncConfig cfg, HiveConf configuration, FileSystem fs) {
   }
 
   public void syncHoodieTable() {
-    LOG.info("Trying to sync hoodie table" + cfg.tableName + " with base path " + hoodieHiveClient
+    switch(hoodieHiveClient.getTableType()) {
+      case COPY_ON_WRITE:
+        syncHoodieTable(false);
+        hoodieHiveClient.close();
+        break;
+      case MERGE_ON_READ:
+        //sync a RO table for MOR
+        syncHoodieTable(false);
+        String originalTableName = cfg.tableName;
+        cfg.tableName = cfg.tableName + SUFFIX_REALTIME_TABLE;
```

Comment:
Another way to do this is to add an optional config parameter to HiveSyncConfig (only for MOR type of tables) and then refactor parts of HiveSyncTool and HiveHoodieClient to use it conditionally for MOR table types. I feel this keeps conditional clauses away from HiveHoodieClient,HiveSyncTool.